src/super_gradients/module_interfaces/module_interfaces.py

```diff
@@ -17,8 +19,7 @@ def get_dataset_preprocessing_params(self):
         ...
 
 
-@runtime_checkable
-class HasPredict(Protocol):
+class HasPredict:
```

Comment:
What is the issue with `Protocol` ?
Is it because it creates an implicit/hidden contract between classes that implement predict and `HasPredict` ?

Reply:
But why not making it abstract ? I think that if a class inherits from `HasPredict` without implementing the required classes this should raise an exception on instantiation, not on call.

Reply:
The problem with Protocol is that it was messing up [MRO](https://levelup.gitconnected.com/method-resolution-order-in-python-5afbaecc25e0) so that derived methods `get_decoding_module` were not called.
Empirically I found that removing protocol solves the problem. Other methods (changing order or inheritance and playing with explicit call of c'tors for base classes) did not help